service/spotify_service.go

```diff
@@ -0,0 +1,164 @@
+/*
+	Service focus on Searching information about Music Artist using Spotify API.
+	DOCs: https://developer.spotify.com/documentation/general/guides/authorization/client-credentials/
+*/
+
+package service
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/ErickRodriguezWize/academy-go-q42021/domain/model"
+	spotiferr "github.com/ErickRodriguezWize/academy-go-q42021/errors"
+)
+
+var (
+	// Environment Variables.
+	/* CONST */ SpotifyEndpoint = GetEnvVariable("SPOTIFY_API_ENDPOINT")
+	/* CONST */ SpotifyRefreshEndpoint = GetEnvVariable("SPOTIFY_REFRESH_ENDPOINT")
+	/* CONST */ SpotifyAuhtorizationToken = GetEnvVariable("SPOTIFY_AUTHORIZATION_TOKEN")
+	/* CONST */ SpotifyRefreshToken = GetEnvVariable("SPOTIFY_REFRESH_TOKEN")
+	/* CONST */ LimitArtist = GetEnvVariable("LIMIT_ARTIST")
+)
+
+// Struct made using JSON to GO Structure Tool: https://mholt.github.io/json-to-go/
+// ArtistResponseJSON is a struct that handle the "unmarshall" information from HTTP Response Body.
+type ArtistResponseJSON struct {
+	Artists struct {
+		Items []struct {
+			Genres       []string `json:"genres"`
+			SpotifyID    string   `json:"id"`
+			Name         string   `json:"name"`
+			ExternalUrls struct {
+				Spotify string `json:"spotify"`
+			} `json:"external_urls"`
+		} `json:"items"`
+	} `json:"artists"`
+}
+
+//SearcArtist: Makes a HTTP GET call to Spotify API to search for an Artist Information.
+// This func will return an error (in case that one is triggered).
+func SearchArtist(artist string, targetArtist *model.Artist) error {
+	//Get token from Spotify API (Token experies every 30 minutes.)
+	accessToken, err := RefreshToken()
+	if err != nil {
+		return spotiferr.TokenMissing
+	}
+
+	artist = strings.ReplaceAll(artist, "+", " ")
+
+	//Encoded spaces with "%20". Spotify ask for this escaping for correctly HTTP Calls.
+	encoded_artist := url.QueryEscape(artist)
+	endpoint := SpotifyEndpoint + "search?q=artist:" + encoded_artist + "&type=artist&limit=" + LimitArtist
+	log.Println("*** HTTP GET to", endpoint)
+
+	//Create request with URL endpoint, Method and Headers.
+	request, err := http.NewRequest("GET", endpoint, nil)
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("Authorization", "Bearer "+accessToken)
+	if err != nil {
+		return spotiferr.BadRequestFormat
+	}
```

Comment:
Error check should be performed just after getting the value of it. 
```suggestion
	request, err := http.NewRequest("GET", endpoint, nil)
	if err != nil {
		return spotiferr.BadRequestFormat
	}
	
	request.Header.Set("Content-Type", "application/json")
	request.Header.Set("Authorization", "Bearer "+accessToken)
```